src/run.ts

```diff
@@ -32,17 +33,31 @@ process.on('unhandledRejection', (reason, _promise) => {
 async function runDevChainFromTar(
 	filename: string,
 	port: number,
+	db: string,
 	onStart?: (port: number, stop: () => Promise<void>) => void) {
-	const chainCopy: tmp.DirResult = tmp.dirSync({ keep: false, unsafeCleanup: true })
-	console.log(`Creating tmp folder: ${chainCopy.name}`)
-	await decompressChain(filename, chainCopy.name)
-	const stopGanache = await startGanache(
-		chainCopy.name,
-		{
-			verbose: true,
-			port: port,
-			onStart: onStart,
-		})
+	let stopGanache: () => Promise<void>
+	if (db != undefined) {
```

Comment:
Could more logic be deduplicated between the two branches? Looks like the only difference is that when `db` is undefined, we need to create a temporary directory. Decompressing and starting ganache could be done outside this if-else.